xla/service/gpu/transforms/dynamic_slice_fusion_rewriter.cc

```diff
@@ -152,70 +156,76 @@ bool IsAlignedSlice(const HloInstruction* slice) {
   return true;
 }
 
-// Pattern matches the following IR (generated by `jax.lax.scan`) to check if
-// the offset is a loop iteration number:
-
-// clang-format off
-//   param = (s32[], s32[], s32[16]{0}, s32[16]{0}) parameter(0)
-//   // the index in `gte` has to be the loop iteration index
-//   gte = s32[] get-tuple-element(param), index=0
-//   c0 = s32[] constant(0)
-//   compare = pred[] compare(gte, c0), direction=LT
-//   c_trip_count = s32[] constant(16)
-//   add = s32[] add(gte, c_trip_count)
-//   select = s32[] select(compare, add, gte)
-// clang-format on
-
-bool IsLoopIterationNumber(const HloInstruction& offset) {
-  const HloComputation* parent = offset.parent();
-  if (!parent->IsWhileBodyComputation()) return false;
-
-  // Scan loops trip count must be known at compile time as it iterates over the
-  // leading dimension of the statically shaped input.
-  const HloInstruction* while_instr = parent->WhileCallInstruction();
-  auto config = while_instr->backend_config<xla::WhileLoopBackendConfig>();
-  if (!config.ok() || !config->has_known_trip_count()) return false;
-  int32_t trip_count = config->known_trip_count().n();
-
-  // First lets check the offset computation pattern
-  if (!Match(&offset, m::Select(m::Lt(m::GetTupleElement(m::Parameter(0)),
-                                      m::ConstantScalar<int32_t>(0)),
-                                m::Add(m::GetTupleElement(m::Parameter(0)),
-                                       m::ConstantScalar(trip_count)),
-                                m::GetTupleElement(m::Parameter())))) {
+// Returns true if the `consumer` only depends on the `producer` and no other
+// instructions. This is a recursive function checking all paths from the
+// `consumer` to the parameters of the computation and if there is any path
+// without `producer`, then it returns false.
+bool is_only_dependent_on(const HloInstruction* consumer,
+                          HloInstruction* producer,
+                          bool allow_replica_partition_id = false) {
+  if (consumer == producer) return true;
```

Comment:
As specified in a previous PR, we'd like to have braces around all controlled statements (even single-line). Please fix this here and below.

Reply:
Sorry about that. Missed it here. I think it is all covered now, but please let me know if you spot anywhere I missed it. (I will check if we can we get clang-format/clang-tidy to check for this?)